declarative/codegen/src/main/java/module-info.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Code generation support for Helidon Declarative APIs.
+ */
+module io.helidon.declarative.codegen {
```

Comment:
We should discuss the API stability of this work before it goes in. I would vote for it being `@Incubating` at least initially (in 4.x). And when we add docs and examples we need to be very clear of that.